src/base/sys_info.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2021 4Paradigm
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SRC_BASE_SYS_INFO_H_
+#define SRC_BASE_SYS_INFO_H_
+
+#include "absl/strings/ascii.h"
+#include "absl/strings/match.h"
+#include "absl/strings/numbers.h"
+#include "absl/strings/string_view.h"
+#include "base/status.h"
+
+namespace openmldb::base {
+
+constexpr const char* MEM_TOTAL = "MemTotal";
+constexpr const char* MEM_AVAILABLE = "MemAvailable";
+
+struct SysInfo {
+    uint64_t mem_total = 0;       // unit is kB
+    uint64_t mem_used = 0;        // unit is kB
+    uint64_t mem_available = 0;   // unit is kB
+};
+
+base::Status GetSysMem(SysInfo* info) {
```

Comment:
why not use `sysinfo` directly ? checkout https://man7.org/linux/man-pages/man2/sysinfo.2.html

Reply:
i cannot get the real used memory from sysinfo(). It looks like there is no cached memory